app/assets/javascripts/spree/frontend/solidus_paypal_commerce_platform/buttons.js

```diff
@@ -2,58 +2,13 @@ SolidusPaypalCommercePlatform.renderButton = function(payment_method_id, style)
   paypal.Buttons({
     style: style,
     createOrder: function (data, actions) {
-      return Spree.ajax({
-        url: '/solidus_paypal_commerce_platform/orders',
-        method: 'POST',
-        data: {
-          payment_method_id: payment_method_id,
-          order_id: Spree.current_order_id,
-          order_token: Spree.current_order_token
-        }
-      }).then(function(res) {
-        return res.table.id;
-      })
+      return SolidusPaypalCommercePlatform.postOrder(payment_method_id);
```

Comment:
_Continuing from https://github.com/nebulab/solidus_paypal_commerce_platform/pull/35/files#r444394574_

Looks like a loading order problem, this file is probably loaded before those functions are set, by using the surrounding functions we postpone the lookup to a moment when `button_actions.js` is loaded as well, a `//= require ./button_actions` at the top of the file should be sufficient. That said I'm ok with this solution as well, as it might allow more flexibility in case someone wants to customise any of the button-actions.

Reply:
So, the problem with this was that I was writing the line as 
`createOrder: SolidusPaypalCommercePlatform.postOrder(payment_method_id),`
which of course ran the function on initializing, returned it's response, and set that as the result of `createOrder` - which is now not a function. 🤦  Instead, I wrote it like this, which solves the issue:
`createOrder: SolidusPaypalCommercePlatform.postOrder.bind(null, payment_method_id),`

This change makes the buttons.js file a lot cleaner, I'll push the updated version in a second. I also removed a few unnecessary returns.